app/eq.py

```diff
@@ -0,0 +1,268 @@
+            "period_str": self._collex_user_description,
+            "period_id": self._collex_period_id,
+            "form_type": self._form_type,
+            "collection_exercise_sid": self._collex_id,
+            "ru_ref": self._sample_unit_ref + self._checkletter,
+            "ru_name": self._ru_name,
+            "survey_id": self._survey_ref,
+            "case_id": self._case_id,
+            "case_ref": self._case_ref,
+            "account_service_url": self._account_service_url,
+            "trad_as": self._trading_as,
+            "region_code": self._region_code,
+            "language_code": self._language_code
+        }
+
+        # Add any non null event dates that exist for this collection exercise
+        self._payload.update(
+            [(key, value) for key, value in self._collex_event_dates.items() if value is not None]
+        )
+
+        logger.info(payload=self._payload)
+
+        return self._payload
+
+    async def _make_request(self, request: Request):
+        method, url, func = request
+        logger.info(f"Making {method} request to {url} and handling with {func}")
+        async with self._app.http_session_pool.request(
+            method, url, auth=self._app["COLLECTION_INSTRUMENT_AUTH"]
+        ) as resp:
+            func(resp)
+            return await resp.json()
+
+    async def _get_survey(self):
+        url = self._survey_url.format(self._app["SURVEY_URL"], self._survey_id)
+        resp = await self._make_request(Request("GET", url, handle_response))
+        return resp
+
+    async def _get_collection_instrument(self):
+        url = self._ci_url.format(self._app["COLLECTION_INSTRUMENT_URL"], self._ci_id)
+        resp = await self._make_request(Request("GET", url, handle_response))
+        return resp
+
+    async def _get_collection_exercise(self):
+        url = self._collex_url.format(self._app["COLLECTION_EXERCISE_URL"], self._ci_id)
+        resp = await self._make_request(Request("GET", url, handle_response))
+        return resp
+
+    async def _get_collection_exercise_events(self):
+        url = self._collex_url.format(self._app["COLLECTION_EXERCISE_URL"], self._ci_id) + "/events"
+        resp = await self._make_request(Request("GET", url, handle_response))
+        return resp
+
+    async def _get_collex_event_dates(self):
+        return {
+            "ref_p_start_date": find_event_date_by_tag(
+                "ref_period_start", self._collex_events, self._collex_id, True
+            ),
+            "ref_p_end_date": find_event_date_by_tag(
+                "ref_period_end", self._collex_events, self._collex_id, True
+            ),
+            "employment_date": find_event_date_by_tag(
+                "employment", self._collex_events, self._collex_id, False
+            ),
+            "return_by": find_event_date_by_tag(
+                "return_by", self._collex_events, self._collex_id, True
+            ),
+        }
+
+    def _get_party_by_business_id(self):
```

Comment:
Shouldn't this be a coroutine? async def?

Reply:
@tej99 still all wip. Sorry for missing off the labels. We wanted to get the automated builds running for docker, etc.